pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -137,20 +149,28 @@ type ArgumentSpec struct {
 // SourceSpec defines the input to the Build
 type SourceSpec struct {
 	// Git represents source in a Git repository.
+	// +optional
 	Git *GitSourceSpec `json:"git,omitempty"`
 
 	// GCS represents source in Google Cloud Storage.
+	// +optional
 	GCS *GCSSourceSpec `json:"gcs,omitempty"`
 
 	// Custom indicates that source should be retrieved using a custom
 	// process defined in a container invocation.
+	// +optional
 	Custom *corev1.Container `json:"custom,omitempty"`
 
 	// SubPath specifies a path within the fetched source which should be
 	// built. This option makes parent directories *inaccessible* to the
 	// build steps. (The specific source type may, in fact, not even fetch
 	// files not in the SubPath.)
+	// +optional
 	SubPath string `json:"subPath,omitempty"`
+
+	// Name is the path under which source will be mounted.
```

Comment:
So this means that if I specify `source: {name: foo, ...}` then my steps can expect there to be a foo directory in `$PWD` aka `/workspace`?

nit: consider something like `// Name is the subpath in workspace where the source will be copied`. It doesn't seem like "mount" is the correct word here as all sources are copied into the workspace volumeMount, rather than a unique volumeMount per source.